primitives/numeric/src/lib.rs

```diff
@@ -212,6 +212,22 @@ impl From<u64> for Numeric {
     }
 }
 
+impl TryFrom<Numeric> for u32 {
+    type Error = <u32 as TryFrom<Decimal>>::Error;
```

Comment:
We are leaking implementation details here.

Reply:
good point

Reply:
fixed